protocol/operator-fee-scalar-fix.md

```diff
@@ -0,0 +1,95 @@
+as information needed to understand the problem and design space.
+If more information is needed on the costs of the problem,
+this is a good place to that information. -->
+
+This is the current operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed / 1e6
+```
+
+Where `operatorFeeScalar` is a `u32`, and `operatorFeeConstant` is a `u64`. However, for a 2.5 million gas transaction and a `operatorFeeScalar` of `u32::MAX_VALUE`, the operator fee will be `u32::MAX_VALUE * 2.5e6 / 1e6 ~= 10 gwei`. With an ETH price of 2.5k, this is 0.0000027 USD -- a negligible fee that is not useful for operators.
+
+# Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+We propose the following change to the operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed * 100
+```
+
+Where `operatorFeeScalar` and `operatorFeeConstant` remain `u32` and `u64` respectively.
+
+Examine the following to get a sense of the range of fees this formula supports. For the dollar cost column, we assume an ETH price of 2.5k.
+
+| Gas Used | `operatorFeeScalar` | `operatorFeeScalar * gasUsed * 100` | Dollar Cost |
+|----------|-------------------|----------------------------------|-------------|
+| 30,000,000 | 1 | 3 gwei | $0.0000075 |
+| 30,000,000 | 2^32 | 1.29e10 gwei | $32,250.00 |
+| 1,000,000 | 1 | 0.1 gwei | $0.00000025 |
+| 1,000,000 | 2^32 | 4.29e8 gwei | $1,072.50 |
+
+Given that the current scalar component fees are negligible, all `operatorFeeScalar` values are set to 0 upon upgrade. Then, chain operators will be responsible for adjusting them according to the new formula. The `operatorFeeConstant` will not change upon transition.
+
+## Scope of changes
+
+This fix will require changes to the `GasPriceOracle` contract as well as execution clients like `op-geth`, in order to reflect the new operator fee formula.
+
+## Alternatives considered
+
+Instead of changing the scalar value, it is also possible to address this problem by changing the `operatorFeeScalar` to a `u64`. With this solution, we would need to add a new field to the `L1Attributes` transaction, deprecate the `setOperatorFeeScalars` function in the `SystemConfig` contract, and add a new function `setJovianOperatorFeeScalars`. Due to the increased complexity, this solution is not preferred.
+
+# Risks & Uncertainties
+
+<!-- An overview of what could go wrong.
+Also any open questions that need more work to resolve. -->
+
+If anyone is using the `operatorFeeScalar` in production, automatically setting it to 0 upon upgrade may cause issues. Since the fee is so low right now, it is likely that any fees lost will be negligible.
```

Comment:
If we are certain no one is using operator fee in production, it might be possible to make the change as a soft fork and discard the old consensus code. Do you have thoughts on that?

Reply:
I agree, this is an interesting question also for Product. If noone is using this feature right now, because it's essentially useless in its current form, we may as well just fix it outside of a fork.

Reply:
Design review meeting, we aligned on the idea that we should ship this as a regular fork. If any chain wants to test early, they will be able to do that even with an incomplete (Jovian) feature set. Not recommended for production networks.